### PR TITLE
Settings refactor step one #755

### PR DESCRIPTION
**Pull request by umgelurgel**
* Replace `os.environ` calls with `decouple.config` calls that handle type casts.
* Replace `sys.path` operations with `unipath.Path` to make the code more readable
* Reorder some of the settings code - move all imports to the top, move code that's not variable assignments or function definitions to the bottom.

### CODE REVIEW

**Comment by umgelurgel** [comment]
`decouple.config` raises an exception if a given variable is not present in a config file or an environmental variable - they currently have default values assigned, but requiring them to be defined locally would be a cleaner option. I can add a `.env` file that would be copied into the project root by the `provision.sh` script.

File: pycon/settings.py [hunk 1]
```diff
@@ -1248,6 +1230,15 @@ def cron_cleanup():
 # This is used just for tests
 DISABLE_CACHING = False
 
+
+### Override any settings with local settings
```
Comment (artcz) [inline pycon/settings.py]: this comment is misplaced

File: pycon/settings.py [hunk 2]
```diff
@@ -4,16 +4,18 @@
 import os.path
 import sys
 
+from decouple import config
+from unipath import Path
```
Comment (artcz) [inline pycon/settings.py]: if we're adding some path management libraries, I'd got with pathlib instead of unipath, especially that its included in python3 (and that it has overloaded '/' which looks better than .child()).

that said – I'd put that in a separate PR anyway and keep this one just about `config()` vs `os.environ`

File: pycon/settings.py [hunk 3]
```diff
@@ -38,39 +40,28 @@
...
-
 ADMINS = (
     ('web-wg', 'web-wg@europython.eu'),
 )
-
 MANAGERS = ADMINS
 
-ALLOWED_HOSTS = os.environ.get("ALLOWED_HOSTS", '*').split(',')
-#APPEND_SLASH = False
+ALLOWED_HOSTS = config('ALLOWED_HOSTS', default='*', cast=lambda v: [s.strip() for s in v.split(',')])
```
Comment (artcz) [inline pycon/settings.py:47]: decouple includes `Csv` maybe we can use it here instead of the labmda?

**Review by artcz — CHANGES_REQUESTED** [review]

**Comment by artcz** [comment]
Overall looks good, added a couple comments in the review mostly about paths.

File: pycon/settings.py (hunk 2 above)
Comment (umgelurgel) [reply pycon/settings.py]: Both good points, will undo the path changes.

**Comment by artcz** [comment]
@umgelurgel python3 rebase needed. :)

**Comment by umgelurgel** [comment]
Closing in favour of https://github.com/EuroPython/epcon/pull/852 - for some reason this PR wan't picking up changes in the branch.
